domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1985,14 +2007,16 @@ private void closeIfNonNull(Transaction transaction) {
       } catch (Exception ex) {
         streamResult.fireExceptionHandler(ex);
       }
+      });
     }).exceptionHandler(e -> {
       rowStream.close();
-      closeIfNonNull(transaction);
-      if (!promise.future().isComplete()) {
+      closeIfNonNull(transaction).onComplete((AsyncResult<Void> voidRes) -> {
```

Comment:
```suggestion
      closeIfNonNull(transaction).onComplete(ignore -> {
```
Braces not needed. Type declaration not needed.
The name `ignore` makes clear that we ignore a close failure on purpose.